lib/finitomata.ex

```diff
@@ -442,20 +443,21 @@ defmodule Finitomata do
           end
         else
           err ->
+            state_with_error = struct(state, error: err)
             cond do
               event in @__config_soft_events__ ->
                 Logger.debug("[⚐⥯] transition softly failed " <> inspect(err))
-                {:noreply, state}
+                {:noreply, state_with_error}
```

Comment:
If you accept the rebinding proposed above, this and the two changes below should go away.